### ISSUE TITLE
False positives from References checks with JSON_TABLE statements

### ISSUE THREAD

**Issue by pprkut** [sqlfluff/sqlfluff#6945]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The `JSON_TABLE` function for MySQL/MariaDB creates a table structure from a JSON string. As such it contains also column definitions. The references family of checks detects these column definitions as column uses instead and wrongly reports errors.

### Expected Behaviour

No errors reported

### Single table select

```sql
set @json = '
[
  {"name":"Laptop", "color":"black", "price":"1000"},
  {"name":"Jeans",  "color":"blue"}
]';

select
    `jt`.`name`,
    `jt`.`color`,
    `jt`.`price`
from json_table(
    @json,
    '$[*]'
    columns(
        `name` varchar(10) path '$.name',
        `color` varchar(10) path '$.color',
        `price` decimal(8, 2) path '$.price'
    )
) as `jt`;
```

```
L:  15 | P:   9 | RF03 | Unqualified reference '`name`' found in single table
                       | select. [references.consistent]
L:  15 | P:   9 | RF03 | Unqualified reference '`name`' found in single table
                       | select which is inconsistent with previous references.
                       | [references.consistent]
L:  16 | P:   9 | RF03 | Unqualified reference '`color`' found in single table
                       | select. [references.consistent]
L:  17 | P:   9 | RF03 | Unqualified reference '`price`' found in single table
                       | select. [references.consistent]
```

### Select with Join

```sql
SET @json = '
[
  {"name":"Laptop", "color":"black", "price":"1000"},
  {"name":"Jeans",  "color":"blue"}
]';

SELECT
    `jt`.`name`,
    `sub`.`size`
FROM (
    SELECT `items`.`size`
    FROM `items`
) AS `sub`
CROSS JOIN JSON_TABLE(
    @json,
    '$[*]'
    COLUMNS(`name` VARCHAR(50) PATH '$.name')
) AS `jt`
```

```
L:  17 | P:  13 | RF02 | Unqualified reference '`name`' found in select with more
                       | than one referenced table/view.
                       | [references.qualification]
```

### How to reproduce

Either `sqlfluff lint --dialect=mariadb` or `sqlfluff lint --dialect=mysql` produces above output

### Dialect

MySQL/MariaDB

### Version

```
sqlfluff, version 3.4.0
```
```
Python 3.12.11
```

### Configuration

No specific configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by jorgeparavicini** [comment]
Same behaviour on ClickHouse JSON type columns

**Comment by piatekmarcin** [comment]
Same behavior in oracle dialect
